pkg/lib/cockpit-components-plot.jsx

```diff
@@ -0,0 +1,521 @@
+        // https://www.patternfly.org/v4/guidelines/colors-for-charts/
+        // Hand-selected to have contrast against each other
+        // for all forms for colorblind and full-spectrum people
+        const colors = [
+            '#4d5258', // --pf-chart-color-black-500
+            '#009596', // --pf-chart-color-cyan-300
+            '#a30000', // --pf-chart-color-red-200
+            '#ec7a08', // --pf-chart-color-orange-300
+            '#f4c145', // --pf-chart-color-gold-300
+            '#b2b0ea', // --pf-chart-color-purple-100
+            '#73c5c5', // --pf-chart-color-cyan-200
+            '#bde2b9', // --pf-chart-color-green-100
+            '#0066cc', // --pf-chart-color-blue-300
+        ];
+
+        const paths = [];
+        for (let i = chart_data.length - 1; i >= 0; i--)
+            paths.push(path(chart_data[i].data, colors[i % colors.length], chart_data[i].name || true));
+
+        function start_dragging(event) {
+            if (event.button !== 0)
+                return;
+
+            const bounds = container_ref.current.getBoundingClientRect();
+            const x = event.clientX - bounds.x;
+            if (x >= m_left && x < w - m_right)
+                setSelection({ start: x, stop: x, left: x, right: x });
+        }
+
+        function drag(event) {
+            const bounds = container_ref.current.getBoundingClientRect();
+            let x = event.clientX - bounds.x;
+            if (x < m_left) x = m_left;
+            if (x > w - m_right) x = w - m_right;
+            setSelection({
+                start: selection.start, stop: x,
+                left: Math.min(selection.start, x), right: Math.max(selection.start, x)
+            });
+        }
+
+        function stop_dragging() {
+            const left = x_value(selection.left) / 1000;
+            const right = x_value(selection.right) / 1000;
+            plot_state.zoom_state.zoom_in(right - left, right);
+            setSelection(null);
+        }
+
+        function cancel_dragging() {
+            setSelection(null);
+        }
+
+        // This is a thin transparent rectangle placed at the x-axis,
+        // on top of all the graphs.  It prevents bogus hover events
+        // for parts of the graph that are zero or very very close to
+        // it.
+        const hover_guard =
+            <rect x={0} y={h - m_bottom - 1} width={w} height={5} fill="transparent" />;
+
+        return (
+            <svg width={w} height={h}
+                 ariaLabel={title}
+                 role="img"
+                 onMouseDown={plot_state.zoom_state && plot_state.zoom_state.enable_zoom_in ? start_dragging : null}
+                 onMouseUp={selection ? stop_dragging : null}
+                 onMouseMove={selection ? drag : null}
+                 onMouseLeave={cancel_dragging}>
+                <title>{title}</title>
+                <text x={0} y={-20} style={{ fontSize: "small" }} ref={measure_ref}>{config.widest_string}</text>
```

Comment:
Isn't that inline styles which our CSP will block?

Reply:
CSP doesn't apply to React, no?

Reply:
Not directly, but I would expect that the HTML generated from this would also have a 'style' attribute.